### PR TITLE
[BUG] Modified components to use `dataset_uuid` as identifier

### LINKED ISSUE #153: [BUG] Duplicate datasets in OpenNeuro are selected simultaneously when one is checked
To replicate:

Query for a "Depressive disorder" diagnosis. Among the results will be two datasets with the same name ("Two sessions of resting state with closed eyes..."), and checking the box for one selects both.

These two datasets are 'true' duplicates in OpenNeuro, i.e. they have separate dataset IDs (`dsXXXXXX`) and repos. So checking off one dataset should not check the other dataset as well.

### PR DESCRIPTION
**Pull request by rmanaem**
Closes #153 

Changes proposed in this pull request:

- Implemented `datasetId` prop as an identifier for datasets
- Modified `data-cy` attribute of elements to use `datasetId`
- Updated `updateDownloads` method
- Updated `updateDownloadsSelectAll` method
- Modified `generateTSVString` method to use `dataset_uuid` for filtering the `results` array
- Updated tests

## Checklist

- [ ] PR has an interpretable title with a prefix (`[ENH]`, `[BUG]`, `[DOC]`, `[INFRA]`, `[MAINT]`)
- [ ] PR links to Github issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Code is properly formatted

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: components/ResultCard.vue [hunk 1]
```diff
@@ -15,7 +15,7 @@
                   type="checkbox"
                   value=""
                   :checked="isChecked"
-                  :data-cy="`card-${datasetName}-checkbox`"
+                  :data-cy="`card-${datasetId}-checkbox`"
```
Comment (surchs) [inline components/ResultCard.vue]: datasetUUID? to keep it consistent

File: components/ResultCard.vue [hunk 2]
```diff
@@ -74,6 +74,10 @@
 <script>
 export default {
   props: {
+    datasetId: {
```
Comment (surchs) [inline components/ResultCard.vue]: `UUID`.

**Review by surchs — APPROVED** [review]
Look good @rmanaem. One request: let's change the propName to `datasetUUID` to save ourselves some confusion down the road (i.e. that this is about the UUID and not e.g. the OpenNeuro id)

**Comment by surchs** [comment]
And with that good to go :cook:

File: components/ResultCard.vue (hunk 2 above)
Comment (rmanaem) [reply components/ResultCard.vue]: it would have to be `datasetUuid` Vue doesn't seem to like all caps `UUID`

**Pull request merged**